docs/topics/cache.txt

```diff
@@ -496,6 +562,75 @@ flag on the connection's socket::
         }
     }
 
+``Redis cache arguments``
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The redis caching backend accepts the following additional arguments:
+
+* ``pool_class``: Connection pools create a set of connection which are retrieved
+  as and when required. You can set it to the connection pool of your choice or
+  even develop a custom connection pool. This defaults to ``redis.ConnectionPool``::
+
+    CACHES = {
+        'default': {
+            'BACKEND': 'django.core.cache.backends.redis.RedisCache',
+            'LOCATION': 'redis://127.0.0.1:6379',
+            'OPTIONS': {
+                'pool_class': 'redis.BlockingConnectionPool'
+            }
+        }
+    }
+
+* ``db``: By default, Redis ships with 16 logical databases. A db of your choice can
+  be selected either by mentioning it in the ``LOCATION`` as querystring or path
+  argument, or set it in the ``OPTIONS``. The default choice for this is 0.
+  We can set the db in the following manner::
+
+    CACHES = {
+        'default': {
+            'BACKEND': 'django.core.cache.backends.redis.RedisCache',
+            'LOCATION': 'redis://127.0.0.1:6379?db=10', # option 1
+            'LOCATION': 'redis://127.0.0.1:6379/10', # option 2
+            'OPTIONS': {
+                'db': '10' # option 3
+            }
+        }
+    }
+
+    Note: If multiple values of db are given, the first value encountered
+    according to the above order would be taken.
+
+* ``parser_class``: Parser classes can control on how the responses from the Redis
+  server are parsed. Few parsers like ``hiredis`` can give around 10x speed up. The
+  default choice is ``redis.connection.HiredisParser``, if hiredis is installed
+  else it falls back on using the ``redis.connection.PythonParser``::
+
+    CACHES = {
+        'default': {
+            'BACKEND': 'django.core.cache.backends.redis.RedisCache',
+            'LOCATION': 'unix:/var/run/redis/redis.sock',
+            'OPTIONS': {
+                'parser_class': 'redis.connection.HiredisParser'
+            }
+        }
+    }
+
+* ``pickle_protocol``: Redis is capable of storing strings in the form of key-value
+  pairs. Thus we need to serialize the data before storing it and deserialize it
+  while retrieving it. Python's pickle module is used for this task. We can select
+  the pickle protocol used for the serialization. This defaults to -1 (highest).::
+
+    CACHES = {
+            'default': {
+                'BACKEND': 'django.core.cache.backends.redis.RedisCache',
+                'LOCATION': 'rediss://127.0.0.1:6379',
+                'OPTIONS': {
+                    'pickle_protocol': 3 # highest protocol
+                }
+            }
+        }
```

Comment:
In that case, we can remove the `PickleSerializer` for good and simply use the `pickle.dumps` and `pickle.loads` with `protocol=HIGHEST_PROTOCOL` just like in few of the other cache backends.

Reply:
So `OPTIONS` is passed as the arguments to the client, e.g. to `pymemache.HashClient()` for `PyMemcacheCache`:

https://github.com/django/django/blob/54a30a7a00fea6c5e3702282ade6e0238e06de3b/django/core/cache/backends/memcached.py#L40

We generally don't need to document all of those options that can be passed to the underlying client.

However... Things are a little bit different here. We are not using a third-party library to provide the client class which is what we are doing for the memcached backends, but writing our own. As such, none of the options for our client class are documented anywhere. Having said that, I'm not sure we need to document it at this stage. We may decide to do so later if we expose additional functionality.

I think that it still makes sense to keep `PickleSerializer` and not pass in `pickle_protocol` to `OPTIONS`. We should then do the same as for `PyMemcacheCache` and pass some default options:

https://github.com/django/django/blob/54a30a7a00fea6c5e3702282ade6e0238e06de3b/django/core/cache/backends/memcached.py#L237-L242

(Note that `serde` is the "serializer/deserializer" instance.)

 So perhaps we should pass the serializer in a similar way and change `RedisCache.__init__()` to have:
 
```python
        self._options = {'serializer': PickleSerializer(), **params.get('OPTIONS', {})}
```

This would then allow the following configuration:

```python
                'OPTIONS': {
                    'serializer': PickleSerializer(pickle.DEFAULT_PROCOL),  # Use default protocol instead of highest.
                }
```

Or equally a custom serializer, e.g. using `msgpack`, could be provided.

In summary, let's keep it flexible but avoid documenting everything for now. Once documented as public API we can't refine it as easily. Get the basic thing working and merged first and then extend as required.

Reply:
Yes this sounds like a better approach. Making the changes now and leaving the documentation for this for now